rooms.js

```diff
@@ -163,6 +165,108 @@ var Room = (function () {
 		}
 		return true;
 	};
+	//mute handling
+	Room.prototype.runMuteTimer = function () {
+		if (this.muteTimer || this.mutedUsers.length === 0) return;
+		var timeUntilExpire = this.mutedUsers[0].time - Date.now();
+		if (timeUntilExpire <= 0) {
+			var user = Users(this.mutedUsers[0].userid);
+			if (user) {
+				user.popup("Your mute has expired.");
+				this.unmute(user);
+			} else {
+				this.mutedUsers.splice(0, 1);
+				this.runMuteTimer();
+			}
+		} else {
+			var self = this;
+			this.muteTimer = setTimeout(function () {
+				var user = Users(self.mutedUsers[0].userid);
+				if (user) {
+					user.popup("Your mute has expired.");
+					self.unmute(user);
+				} else {
+					self.mutedUsers.splice(0, 1);
+					self.runMuteTimer();
+				}
+			}, timeUntilExpire);
+		}
+	};
+	Room.prototype.isMuted = function (user) {
+		if (!user) return;
+		if (this.mutedUsers) {
+			for (var i = 0; i < this.mutedUsers.length; i++) {
```

Comment:
When `n` is small, it's not very clear what's faster. This should be fine in theory.